docs: document PreRenderedChunk properties including isDynamicEntry and isImplicitEntry (#6256)
rollup/rollup: docs/configuration-options/index.md

}
```

The `PreRenderedChunk` type provides information about the chunk being generated:

- `exports`: The list of exported bindings from the chunk.
- `facadeModuleId`: The module id of the entry point this chunk is a facade for, or `null` if this is not a facade.
- `isDynamicEntry`: `true` if this chunk is only reachable via dynamic `import()` expressions.
- `isEntry`: `true` if this chunk is an entry point (either from the `input` option or emitted via `this.emitFile`).

> - `isImplicitEntry`: `true` if this chunk was emitted with [`implicitlyLoadedAfterOneOf`](../plugin-development/index.md#this-emitfile) set, indicating it will only be loaded after at least one of the specified modules has been loaded.

Reviewer: This is somewhat similar. `implicitlyLoadedAfterOneOf` works as if there is a dynamic import between the modules even though it is not spelled out, but it does not prevent the chunk e.g. from being directly imported. But it allows Rollup to generate more optimized chunks because it knows that this is not an independent entry point.
In other words, this chunk will only ever act as an entry point if the other chunks are already in memory. Maybe

```suggestion
- `isImplicitEntry`: `true` if this chunk was emitted with [`implicitlyLoadedAfterOneOf`](../plugin-development/index.md#this-emitfile) set, indicating it will only be loaded as an entry point if at least one of the specified modules have already been loaded.
```
